### ISSUE TITLE
[Question]  Lamdba values of Loss function

### ISSUE THREAD

**Issue by Teslatic** [carla-simulator/imitation-learning#38]
Hey,

just one question regarding the Loss function (equation 6 in your paper).
What values did you use for the Lambda?

Thanks in advance & cheers

**Comment by felipecode** [comment]
It was: 
 Steer: 0.5
  Gas: 0.45
  Brake: 0.05

**Comment by Teslatic** [comment]
Thank's for your reply.
Unfortunately I don't think that I understood it correctly, because of two reasons:

- There is no lambda in the equation which takes care of the the steering angle
- the "brake" component does not seem to be an output of the model (please correct me if i'm wrong)


What I understood:
```
# brake is 1.0 if acceleration is zero or below, regarding the targets
if brake != 1.0:
  lambda_acc = 0.45
else:
  lambda_acc = 0.05
lambda_steering = 0.5

loss =  lambda_steering * || s - s_gt ||² + lambda_acc * || a - a_gt ||²
```

Could you verify/falsify my assessment?

**Comment by mees** [comment]
@felipecode can you comment on @Teslatic's issue please?

**Comment by felipecode** [comment]
Hello the version provided here is the CARLA paper version  which actually contains brake on the output as well.
So the pseudocode for that would be:
```
lambda_steering = 0.5
lambda_brake = 0.05
lambda_acc = 0.45
loss =  lambda_steering * || s - s_gt ||² + lambda_acc * || a - a_gt ||² + lambda_brake * || b - b_gt ||² 

```
For the previous version we would just merge acceleration with brake, making acceleration to range from -1 to 1.  The weights were equal, 0.5  for acc and 0.5 for steering. 
This merging was producing worse results, specially when adding new weathers as in the CARLA paper.

**Comment by mees** [comment]
@felipecode thanks for the infos, but we are asking about the "End-to-end Driving via Conditional Imitation Learning" paper. In Equation 6 there is no lambda for the steering angle and the lambda for the acceleration is not mentioned.

**Comment by felipecode** [comment]
Sorry, I think I answered in the previous answer no ?

For the previous version (Conditional imitation learning paper) we would just merge acceleration with brake, making acceleration to range from -1 to 1. The weights were equal, 0.5 for acc and 0.5 for steering.

**Comment by mees** [comment]
Oh I see, sorry I thought that the conditional imitation paper was posterior to the carla paper, that's why I didn't know what you were referring to with 'previous paper'. Thanks for the clarification!

**Comment by felipecode** [comment]
You are right that is really confusing. The CARLA version is the newer one, even though the paper was published before.
